packages/plugin-constraints/sources/commands/constraints.ts

```diff
@@ -35,51 +35,62 @@ export default class ConstraintsCheckCommand extends BaseCommand {
 
   @Command.Path(`constraints`)
   async execute() {
-    for (let t = 0; t < 10; ++t) {
-      const configuration = await Configuration.find(this.context.cwd, this.context.plugins);
-      const {project} = await Project.find(configuration, this.context.cwd);
-      const constraints = await Constraints.find(project);
+    const configuration = await Configuration.find(this.context.cwd, this.context.plugins);
+    const {project} = await Project.find(configuration, this.context.cwd);
+    const constraints = await Constraints.find(project);
 
-      const toSave = new Set<Workspace>();
+    const report = await StreamReport.start({
+      configuration,
+      stdout: this.context.stdout,
+    }, async report => {
+      let allSaves = new Set<Workspace>();
+      const errors = new Set<[MessageName, string]>();
 
-      const report = await StreamReport.start({
-        configuration,
-        stdout: this.context.stdout,
-      }, async report => {
+      for (let t = 0, T = this.fix ? 10 : 1; t < T; ++t) {
+        errors.clear();
+
+        const toSave = new Set<Workspace>();
         const result = await constraints.process();
 
-        await processDependencyConstraints(toSave, result.enforcedDependencies, {
+        await processDependencyConstraints(toSave, errors, result.enforcedDependencies, {
           fix: this.fix,
           configuration,
-          report,
         });
 
-        await processFieldConstraints(toSave, result.enforcedFields, {
+        await processFieldConstraints(toSave, errors, result.enforcedFields, {
           fix: this.fix,
           configuration,
-          report,
         });
 
-        for (const workspace of toSave) {
-          workspace.persistManifest();
+        allSaves = new Set([
+          ...allSaves,
+          ...toSave,
+        ]);
+
+        // If we didn't apply any change, or there aren't any errors then we can exit the loop
+        if (toSave.size === 0 || errors.size === 0) {
+          break;
         }
-      });
+      }
 
-      if (report.hasErrors())
-        return report.exitCode();
+      // save all modified manifests
+      await Promise.all([...allSaves].map(workspace => workspace.persistManifest()));
```

Comment:
nit:
```suggestion
      await Promise.all([...allSaves].map(async workspace => {
        await workspace.persistManifest();
      }));
```

Reply:
can i ask why ?

Reply:
The rest of the code follows a similar style, where we prefer to pay some vertical space if it allows to decreases the number of concerns for a single line.